geoh5py/objects/geo_image.py

```diff
@@ -150,42 +157,74 @@ def copy_from_extent(
         """
         Sub-class extension of :func:`~geoh5py.shared.entity.Entity.copy_from_extent`.
         """
-        # todo: save the temp grid in a temp workspace?
-        if self.vertices is None:
-            raise AttributeError("Vertices are not defined.")
-
         if self.image is None:
             warnings.warn("Image is not defined.")
             return None
 
-        # transform the image to a grid
-        grid = self.to_grid2d(parent=parent, mode="RGBA")
-
-        # transform the image
-        grid_transformed = grid.copy_from_extent(
-            extent=extent,
-            parent=parent,
-            copy_children=copy_children,
-            clear_cache=clear_cache,
-            inverse=inverse,
-            from_image=True,
-            **kwargs,
+        if inverse:
+            raise NotImplementedError(
+                "Inverse mask is not implemented yet with images."
+            )
+
+        # 1. find the point where the image and extent intersect
+        clipped_uv, plane = compute_clipped_polygon_uv(
+            self.vertices, extent, eps_plane=1e-5, eps_collinear=1e-6
         )
 
-        if grid_transformed is None:
-            grid.workspace.remove_entity(grid)
-            warnings.warn("Image could not be cropped.")
+        # short-circuit degenerate cases
+        if clipped_uv.shape[0] < 3:
+            warnings.warn("The image and the extent doesn't intersect.")
```

Comment:
```suggestion
```
Otherwise when we batch process a whole workspace, we will get a ton of prints to screen.